content/spaces/locking.textile

```diff
@@ -0,0 +1,215 @@
+
+A member must have been "entered":/spaces/space#enter into the space to acquire a lock.
+
+The following is an example of attempting to acquire a lock:
+
+```[javascript]
+const acquireLock = await space.locks.acquire(id);
+```
+
+The following is an example of passing a set of @attributes@ when trying to acquire a lock:
+
+```[javascript]
+const lockAttributes = new Map();
+lockAttributes.set('component', 'cell-d3');
+const acquireLock = await space.locks.acquire(id, { lockAttributes });
+```
+
+Once a member requests a lock by calling @acquire()@, the lock is temporarily in the "pending state":#states. An event will be emitted based on whether the lock request was successful (a status of @locked@) or invalidated (a status of @unlocked@).
+
+h2(#release). Release a lock
+
+Use the @release()@ method to explicitly release a lock once a member has finished editing the related component.
+
+The following is an example of releasing a lock:
+
+```[javascript]
+await space.locks.release(id);
+```
+
+Releasing a lock will emit a lock event with a "lock status":#states of @unlocked@.
+
+<aside data-type='note'>
+<p>When a member "leaves":/spaces/space#leave a space, their locks are automatically released.</p>
+</aside>
+
+h2(#subscribe). Subscribe to lock events
+
+Subscribe to lock events by registering a listener. Lock events are emitted whenever the "lock state":#states transitions. Use the @subscribe()@ method on the @locks@ namespace of the space to receive updates.
+
+All lock events are @update@ events. When a lock event update is received, UI components can be updated to add and remove visual indications of which member is locking them.
+
+The following is an example of subscribing to lock events:
+
+```[javascript]
+space.locks.subscribe('update', (lock) => {
+  console.log(lock);
+});
+```
+
+The following is an example payload of a lock event:
+
+```json
+{
+  "member": {
+    "clientId": "smango",
+    "connectionId": "hs343gjsdc",
+    "isConnected": true,
+    "profileData": {
+      "username": "Saiorse Mango"
+    },
+    "location": {
+      "slide": "sheet-2",
+      "component": "d-14"
+    },
+    "lastEvent": {
+      "name": "update",
+      "timestamp": 1247525689781
+    }
+  },
+  "request": {
+    "id": "s2-d14",
+    "status": "locked",
+    "timestamp": 1247525689781
+    "attributes": {
+      "componentId": "cell-d14"
+    }
+}
+```
+
+The following are the properties of a lock event payload:
+
+|_. Property |_. Description |_.  Type |
+| member.clientId | The "client identifier":/auth/identified-clients for the member. | String |
+| member.connectionId | The unique identifier of the member's "connection":/connect. | String |
+| member.isConnected | Whether the member is connected to Ably or not. | Boolean |
+| member.lastEvent.name | The most recent "event":/spaces/avatar emitted by the member. Will be one of @enter@, @update@, @leave@ or @remove@. | String |
```

Comment:
should this link to avatar stack?

Reply:
Yeah, as it's from the member object and that describes the event types.